src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt

```diff
@@ -0,0 +1,188 @@
+  private val prisonRepository: PrisonRepository,
+  private val referenceCodeRepository: ReferenceCodeRepository,
+  private val notificationRepository: NotificationRepository,
+  private val locationsInsidePrisonClient: LocationsInsidePrisonClient,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun request(request: RequestVideoBookingRequest, username: String) {
+    when (request.bookingType!!) {
+      BookingType.COURT -> sendCourtRequestEmails(request, username)
+      BookingType.PROBATION -> log.info("TODO - send probation request emails.")
+    }
+  }
+
+  private fun sendCourtRequestEmails(request: RequestVideoBookingRequest, username: String) {
+    appointmentsService.checkCourtAppointments(request.prisoner().appointments, request.prisoner().prisonCode!!)
+
+    val (pre, main, post) = getCourtAppointments(request.prisoner())
+
+    val prison = prisonRepository.findByCode(request.prisoner().prisonCode!!)
+      ?.also { require(it.enabled) { "Prison with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Prison with code ${request.prisoner().prisonCode} not found")
+
+    val court = courtRepository.findByCode(request.courtCode!!)
+      ?.also { require(it.enabled) { "Court with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Court with code ${request.courtCode} not found")
+
+    val hearingType = referenceCodeRepository.findByGroupCodeAndCode("COURT_HEARING_TYPE", request.courtHearingType!!.toString())!!
+    val locations = locationsInsidePrisonClient.getLocationsByKeys(setOfNotNull(pre?.locationKey, main.locationKey, post?.locationKey)).associateBy { it.key }
+    val contacts = contactsService.getContactsForCourtBookingRequest(court, prison, username).allContactsWithAnEmailAddress()
+
+    contacts.mapNotNull { contact ->
+      when (contact.contactType) {
+        ContactType.OWNER -> createCourtOwnerEmail(contact, request, court, prison, hearingType, main, pre, post, locations)
+        ContactType.PRISON -> createCourtPrisonEmail(contact, request, court, prison, contacts, hearingType, main, pre, post, locations)
+        else -> null
+      }
+    }.forEach { email ->
+      sendEmailAndSaveNotification(email)
+    }
+  }
+
+  private fun getCourtAppointments(prisoner: UnknownPrisonerDetails): Triple<Appointment?, Appointment, Appointment?> {
+    return prisoner.appointments.appointmentsForCourtHearing()
+  }
+
+  private fun sendEmailAndSaveNotification(email: Email) {
+    emailService.send(email).onSuccess { (govNotifyId, templateId) ->
+      notificationRepository.saveAndFlush(
+        Notification(
+          email = email.address,
+          govNotifyNotificationId = govNotifyId,
+          templateName = templateId,
+          reason = "Booking request",
+        ),
+      )
+    }.onFailure {
+      log.info("BOOKINGS: Failed to send booking request email for prisoner.")
```

Comment:
Not sure what else could be logged here but we have no context of what failed to send just that something failed.

I am beginning to wonder if we should record failures in the DB as well, not suggesting you do this, just a thought that has been in the back of my mind.